### ISSUE TITLE
Hint about automatic deploy to PyPI

### ISSUE THREAD

**Issue by lelit** [joerick/cibuildwheel#20]
I'm trying to setup automatic wheel deploy from Travis to PyPI for a couple of projects of myne.

Everything goes well, the packages get built within the manylinux1 docker image, and tests pass, but when `twine` tries to upload to PyPI, it fails with

> Uploading pg_query-0.2-cp36-cp36m-manylinux1_i686.whl
> HTTPError: 403 Client Error: Invalid or non-existent authentication information. for url: https://upload.pypi.org/legacy/

I followed the examples and read the available documentation, but as this is my first time at that, it may be that I'm missing the obvious.

My [.travis.yml](https://github.com/lelit/pg_query/blob/master/.travis.yml) sets `TWINE_USERNAME`, and I set the `TWINE_PASSWORD` in the Travis project settings to the value obtained by executing `travis encrypt 'my-pypi-password'` as explained [here](https://docs.travis-ci.com/user/encryption-keys/#Usage). I tried also with `travis encrypt 'TWINE_PASSWORD=my-pypi-password'` with the same result.

What am I doing wrong (or not doing at all)?

Can you either shed some light on me, or point me to whatever place where I can learn more on the process?

Thanks in advance!

**Comment by tgarc** [comment]
@lelit I suppose this could be lots of things. But one really nefarious thing I found out about encrypting passwords on travis-ci was that you have to escape special characters. (see issue [here](https://github.com/travis-ci/dpl/issues/377)). Could that be your issue? Either way, this is definitely twine and/or travis-ci related so you should be debugging those parts, not cibuildwheel.

**Comment by joerick** [comment]
Hey @lelit! Nothing add except to say that I've got a working example repo set up here: https://github.com/joerick/cibuildwheel-autopypi-example .

I set the TWINE_PASSWORD variable in the Travis settings, rather than putting an encrypted key in `.travis.yml`. Could be less error-prone?

> I set the TWINE_PASSWORD in the Travis project settings to the value obtained by executing travis encrypt 'my-pypi-password'

Oh, that might be your problem - if you're typing a password into the Travis project settings, you shouldn't encrypt it.
